src/index.js

```diff
@@ -1,390 +1,603 @@
-      children.push(node)
     }
-  }
+  } else {
+    if (name[0] === "o" && name[1] === "n") {
+      name = name.slice(2).toLowerCase()
 
-  return typeof name === "function"
-    ? name(attributes || {}, children)
-    : {
-        nodeName: name,
-        attributes: attributes || {},
-        children: children,
-        key: attributes && attributes.key
+      if (!element.events) element.events = {}
+
+      element.events[name] = nextValue
+
+      if (nextValue == null) {
+        element.removeEventListener(name, eventProxy)
+      } else if (lastValue == null) {
+        element.addEventListener(name, eventProxy)
       }
-}
+    } else {
+      var nullOrFalse = nextValue == null || nextValue === false
 
-export function app(state, actions, view, container) {
-  var map = [].map
-  var rootElement = (container && container.children[0]) || null
-  var oldNode = rootElement && recycleElement(rootElement)
-  var lifecycle = []
-  var skipRender
-  var isRecycling = true
-  var globalState = clone(state)
-  var wiredActions = wireStateToActions([], globalState, clone(actions))
-
-  scheduleRender()
-
-  return wiredActions
-
-  function recycleElement(element) {
-    return {
-      nodeName: element.nodeName.toLowerCase(),
-      attributes: {},
-      children: map.call(element.childNodes, function(element) {
-        return element.nodeType === 3 // Node.TEXT_NODE
-          ? element.nodeValue
-          : recycleElement(element)
-      })
+      if (
+        name in element &&
+        name !== "list" &&
+        name !== "draggable" &&
+        name !== "spellcheck" &&
+        name !== "translate" &&
+        !isSvg
+      ) {
+        element[name] = nextValue == null ? "" : nextValue
+        if (nullOrFalse) {
+          element.removeAttribute(name)
+        }
+      } else {
+        var ns = isSvg && name !== (name = name.replace(/^xlink:?/, ""))
+        if (ns) {
+          if (nullOrFalse) {
+            element.removeAttributeNS(XLINK_NS, name)
+          } else {
+            element.setAttributeNS(XLINK_NS, name, nextValue)
+          }
+        } else {
+          if (nullOrFalse) {
+            element.removeAttribute(name)
+          } else {
+            element.setAttribute(name, nextValue)
+          }
+        }
+      }
     }
   }
+}
+
+var createElement = function(node, lifecycle, eventProxy, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+        ? document.createElementNS(SVG_NS, node.name)
+        : document.createElement(node.name)
+
+  var props = node.props
+  if (props.onCreate) {
+    lifecycle.push(function() {
+      props.onCreate(element)
+    })
+  }
 
-  function resolveNode(node) {
-    return typeof node === "function"
-      ? resolveNode(node(globalState, wiredActions))
-      : node != null
-        ? node
-        : ""
+  for (var i = 0, length = node.children.length; i < length; i++) {
```

Comment:
Please store the length in a variable 
```suggestion
var len = node.children.length
  for (var i = 0, length = len; i < length; i++) {
```

Reply:
`node.children.length` is stored in the `length` variable, declared alongside `i` in the setup part of the `for` loop. Or do you mean something else?